### PR TITLE
chore(Dockerfile): hadolint best practices

### PR DESCRIPTION
**Pull request by maximee-leroyy**

### CODE REVIEW

**Comment by slovdahl** [comment]
Thanks for the PR @maxime1907!

First of all, this repository is more or less in a permanent maintenance mode, so don't expect too much activity to take place.

Would you mind describing in detail what these best practices are about? I'm personally not familiar with hadolint (https://github.com/hadolint/hadolint I presume?).

**Comment by maximee-leroyy** [comment]
> Thanks for the PR @maxime1907!
> 
> First of all, this repository is more or less in a permanent maintenance mode, so don't expect too much activity to take place.
> 
> Would you mind describing in detail what these best practices are about? I'm personally not familiar with hadolint (https://github.com/hadolint/hadolint I presume?).

Yes this is the right link, its a tool that enforces a set of best practices when writing Dockerfiles.
In this PR i have made these changes:
- pin image versions to their SHA for reproducible builds
- mount cache directories for faster builds
- mount build files instead of using COPY so it does not add another unecessary layer (smaller image)
- do not install recommended apt packages
- pin apt packages for reproducible builds
- optimise final image size by only building wheels inside the build stage
- remove unecessary apt package build-essential (smaller image)
- mount dist and wheels instead of using COPY so it does not add another unecessary layer (smaller image)
- remove apt logs (smaller image)
- rootless image for security measures

File: Dockerfile [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+    poetry export -o requirements.txt \
+    && pip wheel --no-deps --wheel-dir /app/wheels -r requirements.txt \
+    && poetry build --quiet --no-ansi --no-interaction --format=sdist
+
+FROM python:3.10-slim@sha256:9956522e7eafd57e3e7bb4b102d56f02882924019867cd2036c1a7c3ee56b174 AS runtime
+
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    apt-get update -q \
+    && apt-get install -yq --no-install-recommends \
+    git=1:2.39.2-1.1 \
```
Comment (slovdahl) [inline Dockerfile:28]: I'm not sure how Debian(?) normally handle things like this, will this exact version be available forever and ever?

**Review by slovdahl — APPROVED** [review]
Thanks, I learned a lot from this :+1:

File: Dockerfile (hunk 1 above)
Comment (maximee-leroyy) [reply Dockerfile:28]: not sure about this one!

**Comment by slovdahl** [comment]
Released in https://github.com/hiboxsystems/marge-bot/releases/tag/0.13.0.

**Comment by maximee-leroyy** [comment]
> Released in https://github.com/hiboxsystems/marge-bot/releases/tag/0.13.0.

Thanks!
Could you publish the 0.13.0 tag on dockerhub because i dont see it listed here:
- https://hub.docker.com/r/hiboxsystems/marge-bot/tags?page=1&name=0.

**Comment by slovdahl** [comment]
@maxime1907 Thanks for the heads up!

I had forgotten the quirks in the deployment target in the `Makefile`.. :unamused: It got pushed as `hiboxsystems/marge-bot:HEAD` and nothing else. But I removed that tag now and pushed the right tags (`0.13.0`, `stable` and `latest`). Should really get this fully automated instead.

**Pull request merged**